api/types/kustomization.go

```diff
@@ -217,11 +218,25 @@ func (k *Kustomization) FixKustomizationPostUnmarshalling() {
 // FixKustomizationPreMarshalling fixes things
 // that should occur after the kustomization file
 // has been processed.
-func (k *Kustomization) FixKustomizationPreMarshalling() error {
+func (k *Kustomization) FixKustomizationPreMarshalling(fSys filesys.FileSystem) error {
 	// PatchesJson6902 should be under the Patches field.
 	k.Patches = append(k.Patches, k.PatchesJson6902...)
 	k.PatchesJson6902 = nil
 
+	if k.PatchesStrategicMerge != nil {
+		for _, patchStrategicMerge := range k.PatchesStrategicMerge {
+			// check this patch is file path select.
+			if _, err := fSys.ReadFile(string(patchStrategicMerge)); err == nil {
```

Comment:
You don't need to change anything, but just as an FYI, the existing `PatchesStrategicMerge` code differentiates between inline and files like this: https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/builtin/patchstrategicmergetransformer/PatchStrategicMergeTransformer.go#L63
Checking for inlines this way doesn't require the file to exist.